### ISSUE TITLE
bug: long text with max width not truncating in button

### ISSUE THREAD

**Issue by alan-agius4** [ionic-team/ionic-framework#8700]
#### Short description of the problem:

Long text in buttons is not being truncated. I am using `full` and `block` buttons

![image](https://cloud.githubusercontent.com/assets/17563226/19384889/28eb0bc8-920c-11e6-8a36-25250f8890b8.png)

While the button does have the ellipsis, as show below;

``` sass
.button {
    text-align: center;
    text-overflow: ellipsis;
    white-space: nowrap;
}
```

This doesn't work, as the `.button-inner` class has  `display: flex` 
#### What behaviour are you expecting?

Display ellipsis.

![image](https://cloud.githubusercontent.com/assets/17563226/19384896/326da1f6-920c-11e6-9733-925ed7f3bd69.png)

This can be done by setting the display of  `.button-inner` to `inline`. However, I don't really know the reason for it's display being flex.

**Which Ionic Version?** 2 RC.0

**Comment by ncapito** [comment]
I think it's there to help with centering?   Not sure how you all would want to fix it, but  the only way I have been able to fix something like this is to give it another inner child. e.g.

 ``` 

 <button ion-button full block>
   <span class='overflow-me'>button[ion-button][block] with other text</span>
  </button>

```

I then put a class on `.overflow-me` to do the ellipsis.

```
   .overflow-me{
    overflow: hidden;
    text-overflow: ellipsis;
   }
```

**Comment by brandyscarney** [comment]
This is actually pretty complex since buttons are used in so many components. Adding an inner div to buttons would cause issues elsewhere, and changing the display from flex to inline causes issues with buttons containing icons and anchor links styled to look like buttons. I am going to remove this from the milestone for now since the code above by @ncapito works, but I'll revisit this issue.

**Comment by mapsandapps** [comment]
Thank you for submitting this issue.

Starting in [Ionic Framework 7.2](https://github.com/ionic-team/ionic-framework/releases/tag/v7.2.0), we encouraged developers to [opt in](https://github.com/ionic-team/ionic-framework/pull/27547) to text wrapping in buttons by setting `ion-text-wrap`. This is available now if you are on Ionic 7.2+.

As a result of the changes merged in #28682, text wrapping of buttons will be the default behavior starting in Ionic version 8.
